yorm/tests/test_utilities.py

```diff
@@ -96,9 +110,86 @@ def it_requires_a_mapped_class_or_instance():
 
 def describe_match():
 
-    def it_is_not_yet_implemented():
-        with expect.raises(NotImplementedError):
-            utilities.match(Mock)
+    def class_factory(model_class, instance_pile):
+        matches = list(
+            utilities.match(
+                model_class,
+                (lambda kind, key: model_class(kind, key)),
```

Comment:
So, the linter is pointing out that this is equivalent to simply using `model_class` as the factory. i.e. using factory in this case is unnecessary. A few actions that I think would be useful here:

- Update `model_class` to require and additional (non-YORM) argument and test that the factory can be used to   pass in a default value for this required constructor argument.
- Remove the factory value for tests that don't need it and/or add additional tests for calling `match()` without a factory